# A report on my bioinformatics internship journey at *icipe*
## Introduction
This report summarises my experience during the bioinformatics incubation and mentorship program at *icipe* during the period of 06/10/2021 to 31/01/2022. The program consists of two months of intensive training and another 2 months of working in a mini-project that is a branch of ongoing genomic activity. The program guarantees an all-rounded experience ranging from soft skills to technical skills such as programming.
I was honoured to be part of the cohort-4 group and privileged to share this experience with my fellow interns.

# Objectives
The program seeks to mould outstanding bioinformaticians through mentorship and technical training and therefore, objectives are specific to an individual depending on the area of expertise one chooses to specialize. During the first week we came up with a roadmap of what we wanted to achieve during the internship, summarised to the following:

* To improve on my programming skills in python and the UNIX shell
* To take part in a genomics project
* To develop soft skills and network with individuals in the field of data science 

# My experience
## Training
The training period, which took place within the first two months of the program, was a wholesome experience, to say the least. It consisted of activities ranging from simple soft skills such as presentation skills to technical training of computer programs and languages such as python and the UNIX shell. It was a rigorous process that was made simple and achievable with the help of a great team at the Molecular Biology and Bioinformatics Unit.
My highlights included an introduction to the bioinformatics concept, the UNIX shell, Nextflow workflows, Machine Learning concepts, soft skills and python language.

## Reproducing a scientific paper
One of the program deliverables was to reproduce a scientific paper assigned to us by the instructor. The task was a collaborative project of two individuals, and we worked with my partner to reproduce the paper, '[A Phylogenetic and Phenotypic Analysis of *Salmonella enterica Serovar Weltevreden*, an Emerging Agent of Diarrheal Disease in Tropical Regions](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4750946/)' 
During this activity, I interacted more with bash scripting, variant calling and phylogenetic and phenotypic analysis.

## Journal club presentation
The MBBU has a journal club that meets every 2 weeks and an individual presents a journal review of their choice. The meetings were one of my biggest highlights because I kept learning new concepts every day and most importantly upskilled my presentation skillset.
I had an opportunity to deliver a presentation on the paper,'[A Long-Term Engagement with a Social Robot for Autism Therapy](https://www.frontiersin.org/articles/10.3389/frobt.2021.669972/full)'

## Mini-Project
During the program, my partner and I were tasked with developing a machine learning algorithm that identifies an insect. To kickstart the project, we had to identify an insect of interest and through consultations, we decided to settle on ticks(not insects but arachnids) a subject of interest at *icipe*.
We did background research on the topic to identify a favourable model convenient in solving image data problems. This included going through previous research done and already developed working models to understand the best approach in developing our deep learning model.
Data availability was a challenge that we overcame by using the feature transfer method - this includes training a model on something similar to the target data  so that when data becomes available it's implemented into that model after a few optimization steps - in our case we used beetles images downloaded from [kaggle.](https://www.kaggle.com/kmldas/insect-identification-from-habitus-images) This allowed us to make great strides in the model implementation by running 4 models and selecting the most optimal for deployment.
The project was full of great learning points and I truly enjoyed each one of them.

## Others
In addition, we also had weekly updates and progress reviews on tasks done during the week. 

# Conclusion
The whole experience has been very impactful both in my career and as an individual. Having the opportunity to be part of a team that strives for nothing less than excellence has pushed me to want nothing less for myself and in everything that I do. My take-home learning points are:
1. The importance of having a clear workflow for any task and documentation of the process followed toward finding the solution
2. Progress is progress no matter how minute
3. There is power in collaboration 

# Additional materials
* These are some of my first GitHub repositories that I created during training : [My first wiki page](https://github.com/CarolKogei/my-first-wiki), [Website](https://github.com/CarolKogei/website) and [concepts of molecular biology](https://github.com/CarolKogei/Concepts-of-molecular-biology)
* A [powerpoint presentation](https://docs.google.com/presentation/d/1RxlyoQmJ4c3YIo-LjAtUhxBdRnLs9OST/edit?usp=sharing&ouid=117239724346945185906&rtpof=true&sd=true) I created for my journal club presentation
* [Reproducing a scientific research paper](https://github.com/CarolKogei/reproducing-phylogenetics-paper)
* Google colab [notebooks](https://github.com/CarolKogei/google-notebooks-dl) we used for our mini-project
* [Bash assignment scripts](https://github.com/CarolKogei/shell-assignment)
* [python](https://github.com/CarolKogei/Joint_Python) assignment

